From 1719e904acf19499209b16a8a008f55390a7b5e2 Mon Sep 17 00:00:00 2001
From: Vincent Ambo <mail@tazj.in>
Date: Sun, 29 Jan 2023 13:36:12 +0300
Subject: [PATCH] feat: always set the List-ID header even in -watch

Without bothering to figure out exactly how this code path is usually
triggered, always set a list ID when ingesting new emails in
public-inbox-watch.
---
 lib/PublicInbox/Watch.pm | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/lib/PublicInbox/Watch.pm b/lib/PublicInbox/Watch.pm
index 3f6fe21..147971c 100644
--- a/lib/PublicInbox/Watch.pm
+++ b/lib/PublicInbox/Watch.pm
@@ -188,6 +188,10 @@ sub _remove_spam {
 sub import_eml ($$$) {
 	my ($self, $ibx, $eml) = @_;
 
+        # TVL-specific: always set the list-id header, regardless of
+        # any of the other logic below.
+        PublicInbox::MDA->set_list_headers($eml, $ibx);
+
 	# any header match means it's eligible for the inbox:
 	if (my $watch_hdrs = $ibx->{-watchheaders}) {
 		my $ok;
-- 
2.39.0

